pytensor/tensor/slinalg.py

```diff
@@ -67,29 +67,52 @@ def make_node(self, x):
     def perform(self, node, inputs, outputs):
         [x] = inputs
         [out] = outputs
-        try:
-            # Scipy cholesky only makes use of overwrite_a when it is F_CONTIGUOUS
-            # If we have a `C_CONTIGUOUS` array we transpose to benefit from it
-            if self.overwrite_a and x.flags["C_CONTIGUOUS"]:
-                out[0] = scipy_linalg.cholesky(
-                    x.T,
-                    lower=not self.lower,
-                    check_finite=self.check_finite,
-                    overwrite_a=True,
-                ).T
-            else:
-                out[0] = scipy_linalg.cholesky(
-                    x,
-                    lower=self.lower,
-                    check_finite=self.check_finite,
-                    overwrite_a=self.overwrite_a,
-                )
 
-        except scipy_linalg.LinAlgError:
-            if self.on_error == "raise":
-                raise
-            else:
+        # Quick return for square empty array
+        if x.size == 0:
+            eye = np.eye(1, dtype=x.dtype)
+            (potrf,) = scipy_linalg.get_lapack_funcs(("potrf",), (eye,))
+            c, _ = potrf(eye, lower=False, overwrite_a=False, clean=True)
+            out[0] = np.empty_like(x, dtype=c.dtype)
```

Comment:
```suggestion
            (potrf,) = scipy_linalg.get_lapack_funcs(("potrf",), (x,))
            out[0] = np.empty_like(x, dtype=potrf.dtype)
```

Since you just want the dtype, you can check `potrf.dtype`, you don't need to actually call the routine. It's also fine to pass in the empty `x` in this case, since it's only used for its dtype attribute (the data or lack thereof doesn't matter)